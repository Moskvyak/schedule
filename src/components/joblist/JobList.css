.joblist__wrapper {
    padding: 20px;
}