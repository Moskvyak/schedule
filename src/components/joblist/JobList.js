import React from 'react';
import PropTypes from 'prop-types';

import { JobListItem } from '../joblist-item/JobListItem';

import './JobList.css';

export const JobList = ({ options, title }) =>
  <div className="joblist__wrapper">
    {title && <h3>
      {title}
    </h3>}
    {options.map(option => <JobListItem key={option.id} option={option} />)}
  </div>;


JobList.propTypes = {
  options: PropTypes.array.isRequired,
  title: PropTypes.string
}