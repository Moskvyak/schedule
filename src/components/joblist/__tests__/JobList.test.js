import React from 'react';
import renderer from 'react-test-renderer';
import { JobList } from '../JobList';

describe('<JobList />', () => {
  const requiredProps = {
    options: [
      {
        id: '1',
        name: 'test',
        start: '2018-09-01T11:00:00Z',
        end: '2018-09-01T12:00:00Z',
        contact: { name: 'testName' }
      },
      {
        id: '2',
        name: 'test2',
        start: '2018-09-01T11:00:00Z',
        end: '2018-09-01T12:00:00Z',
        contact: { name: 'testName2' }
      }
    ]
  };

  it('should render with required props', () => {
    const component = renderer.create(<JobList {...requiredProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
