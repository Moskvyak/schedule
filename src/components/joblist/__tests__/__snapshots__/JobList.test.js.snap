// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobList /> should render with required props 1`] = `
<div
  className="joblist__wrapper"
>
  <div
    className="joblist-item__wrapper"
  >
    <div
      className="joblist-item__title"
    >
      <span>
        testName
      </span>
       - 
      test
    </div>
    <div
      className="joblist-item__time-period"
    >
      09/01/2018 11:00 AM
       - 
      09/01/2018 12:00 PM
    </div>
  </div>
  <div
    className="joblist-item__wrapper"
  >
    <div
      className="joblist-item__title"
    >
      <span>
        testName2
      </span>
       - 
      test2
    </div>
    <div
      className="joblist-item__time-period"
    >
      09/01/2018 11:00 AM
       - 
      09/01/2018 12:00 PM
    </div>
  </div>
</div>
`;
